### ISSUE TITLE
SELU Activations

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#601]
https://arxiv.org/abs/1706.02515

A number of early users have claimed significant boost in RL applications using the new SELU. Might be good for us to add in an implementation (think it's only a few lines of code).

**Comment by cxhernandez** [comment]
👍 

improved training time for the molecular autoencoder

**Comment by rbharath** [comment]
Props to @vipulraheja for the implementation in #603! Keeping this thread open for future contributions adding Alpha-dropout and some unit tests.

**Comment by vipulraheja** [comment]
Thanks @rbharath for the help and quick review! I had actually started working on both the following tasks, but ran into a bunch of tensorflow setup issues (seg faults). Hopefully should be able to put up a PR in the coming week(s).

**Comment by patrickhop** [comment]
(FYI) From my experiments, removing batch normalization in favor of SELU on graph convolutions improves the convergence rate but doesn't really move the needle performance wise.

**Comment by patrickhop** [comment]
FYI: selu requires 1.0 / sqrt(fan_in) initialization. I think we're using 2 / sqrt(fan_in + fan_out).

**Comment by rbharath** [comment]
Bugfix PRs welcome :)

**Comment by peastman** [comment]
Are you saying that SELU is implemented incorrectly, or just that particular models aren't being initialized in the correct way to work with it?  If the latter, which ones?

**Comment by patrickhop** [comment]
It looks like SELU is implemented correctly, but if you want to throw it on top of something, there are a lot of mathematical assumptions conditioned on what's underneath it for the activations to indeed converge to zero mean and unit variance. The alpha and scale constants in the deepchem codebase are derived for fully connected networks initialized with 1.0 / sqrt(fan_in) (gaussian init).

Right now, we're using uniform glorot initialization in tensorgraph graphconv. I don't know about the other ones.

**Comment by rbharath** [comment]
Closing this since #603 has been merged in.
